File: app/src/main/java/com/fractalwrench/crazycats/mocks/MockFailureRepository.java
package com.fractalwrench.crazycats.mocks;

import android.support.annotation.NonNull;

import com.fractalwrench.crazycats.image.data.ImageData;
import com.fractalwrench.crazycats.image.data.ImageDataRepository;

import java.util.List;

import io.reactivex.Observable;

/**
 * A repository which is guaranteed to produce an error for every call, used for testing.
 */
public class MockFailureRepository implements ImageDataRepository {

    @Override
    public Observable<List<ImageData>> fetchImageSummaries() {
        return Observable.error(new RuntimeException());
    }

    @Override
    public Observable<ImageData> fetchImageById(@NonNull String id) {
        return Observable.error(new RuntimeException());
    }

}

File: app/src/main/java/com/fractalwrench/crazycats/mocks/MockFailureRepository.kt
package com.fractalwrench.crazycats.mocks

import com.fractalwrench.crazycats.image.data.ImageData
import com.fractalwrench.crazycats.image.data.ImageDataRepository

import io.reactivex.Observable

/**
 * A repository which is guaranteed to produce an error for every call, used for testing.
 */
class MockFailureRepository : ImageDataRepository {

    override fun fetchImageSummaries(): Observable<List<ImageData>> {
        return Observable.error(RuntimeException())
    }

    override fun fetchImageById(id: String): Observable<ImageData> {
        return Observable.error(RuntimeException())
    }

}
